### PR TITLE
Validation model specs

### PR DESCRIPTION
**Pull request by serengia**
Hello. Kindy review my request to merge `validation-model-specs` branch to `development` branch. 
The branch contains the implementation of the following:-
- Validation for User, Post, Comment, and Like models
- Testing for User, Post, Comment, and Like models

### CODE REVIEW

File: spec/model/like_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    author: user,
+    comments_counter: 0,
+    likes_counter: 0
+  )
+
+  describe 'associations' do
+    it 'belongs to an author (User)' do
+      association = described_class.reflect_on_association(:author)
+      expect(association.macro).to eq(:belongs_to)
+      expect(association.foreign_key).to eq(:author_id)
```
Comment (t-yanick) [inline spec/model/like_spec.rb]: - [x] Great job adding tests. As per the screenshot below, some 3 tests are failing. Kindly see details on the screenshot and fix the failing tests.
image

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @serengia , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  User and Post model have all required validations
:heavy_check_mark: Unit tests for User, Post, Comment, and Like model present
:heavy_check_mark: Use of correct Gitflow

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ] Great job working on this milestone. Deployment is not a requirement for this milestone. At the moment you have a [live demo link](https://github.com/serengia/rails-blog-app/tree/validation-model-specs) in your README and when clicked, it redirects to your repo. Kindly delete this section or allow just the text `Coming Soon....`

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/model/like_spec.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@
     it 'belongs to an author (User)' do
       association = described_class.reflect_on_association(:author)
       expect(association.macro).to eq(:belongs_to)
-      expect(association.foreign_key).to eq(:author_id)
+      expect(association.foreign_key).to eq(author_id)
       expect(association.class_name).to eq('User')
     end
```
Comment (elisha2kyakpo1) [inline spec/model/like_spec.rb]: - [ ] Great job working on the failing tests, but the issue is still available. I would advise you to empty the database when running the tests, perhaps you are testing using data in your database.

test

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @serengia, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

👍 You did well working on the readme file making it look professinal.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/model/like_spec.rb (hunk 2 above)
Comment (KDlamini) [reply spec/model/like_spec.rb]: Well resolved ✔️ 

image

File: README.md
Comment (KDlamini) [inline README.md:1]: [OPTIONAL] For a professional README kindly make sure that the License suggestion is removed. It is meant to recommend a licence type you can use, it's not meant to be part of the final documentation.

image

**Review by KDlamini — APPROVED** [review]
### Hi @serengia 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Professional pull request
✔️ Model validations for Users and Posts are present
✔️ Validation specs are well-written

# Status: Approved ✔️ 

LeonardoInGatsby

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**